src/plugins/discover/public/components/data_types/logs/log_level_badge_cell.tsx

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { EuiBadge, mathWithUnits, useEuiTheme } from '@elastic/eui';
+import type { CSSObject } from '@emotion/react';
+import { getLogLevelCoalescedValue, getLogLevelColor } from '@kbn/discover-utils';
+import { euiThemeVars } from '@kbn/ui-theme';
+import type { DataGridCellValueElementProps } from '@kbn/unified-data-table';
+import React from 'react';
+
+const badgeCss: CSSObject = {
+  marginTop: '-4px',
+  maxWidth: mathWithUnits(euiThemeVars.euiSize, (size) => size * 7.5),
+};
+
+export const getLogLevelBadgeCell =
+  (logLevelField: string) => (props: DataGridCellValueElementProps) => {
+    const { euiTheme } = useEuiTheme();
+    const value = props.row.flattened[logLevelField];
+
+    if (!value) {
+      return <span data-test-subj="logLevelBadgeCell-empty">-</span>;
+    }
+
+    const coalescedValue = getLogLevelCoalescedValue(value);
+    const color = coalescedValue ? getLogLevelColor(coalescedValue, euiTheme) : undefined;
```

Comment:
```suggestion
    const color = coalescedValue ? getLogLevelColor(coalescedValue, euiTheme) : undefined;
    
    if (!color || !coalescedValue) {
      return <span data-test-subj="logLevelBadgeCell-unknown">{value}</span>
    }
```

Maybe we should render it as `EuiBadge` only for known values and as simple text for others?

Reply:
Current example:

<img width="537" alt="Screenshot 2024-07-16 at 16 33 06" src="https://github.com/user-attachments/assets/4e414e90-f0b9-4c18-a155-fdaf23088086">